torch/csrc/jit/codegen/cuda/lower_unroll.cpp

```diff
@@ -73,6 +73,16 @@ void UnrollPass::handle(kir::Expr* expr) {
         ? ir_builder.trueVal()
         : GpuLower::current()->threadPredMap().getPredicate(out_tv->fuserTv());
 
+    // When this expr is in an unswitched block, only attach the
+    // thread predicate to the expr as thread predicates are not
+    // grouped to the unswitch predicate.
+    kir::Predicate* thread_pred_expr = nullptr;
+    if (unswitched_loop_) {
+      thread_pred_expr = ir_builder.create<kir::Predicate>(thread_pred);
+    }
+
+    non_trivial_pred_found_ = true;
```

Comment:
Is this always true now? I'm confused what this flag protects against now. Is it just scalar ops?